agent/taskresource/cgroup.go

```diff
@@ -0,0 +1,225 @@
+// +build linux
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package taskresource
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+	"github.com/cihub/seelog"
+	"github.com/containerd/cgroups"
+	"github.com/pkg/errors"
+)
+
+const (
+	memorySubsystem         = "/memory"
+	memoryUseHierarchy      = "memory.use_hierarchy"
+	rootReadOnlyPermissions = os.FileMode(400)
+)
+
+var (
+	enableMemoryHierarchy = []byte(strconv.Itoa(1))
+)
+
+// CgroupResource represents Cgroup resource
+type CgroupResource struct {
+	control             cgroup.Control
+	CgroupRoot          string
+	CgroupMountPath     string
```

Comment:
Does this need to be part of `CgroupResource` object? This should be just read from the config object, no?

Reply:
Yes, this should be read from the config object and passed while creating the `CgroupResource`object. In that case, we can directly use the `CgroupResource`'s `CgroupMountPath ` while `Create()`. Else, we might have to pass the config object to the `Create()` resource method, which might be unnecessary for other resources. 
The idea is to make the resource object mostly self-sufficient to create cgroups itself. Thoughts?

Reply:
I prefer this as written  -- this module is decoupled from the config package.  Its easier to test and we don't hold a pointer to global state.